server: Revert server.shutdown.jobs_wait to 0

miretskiy opened the pull request:
Revert default setting for `server.shutdown.jobs_wait` to 0
to ensure that shutdown dows not wait for active jobs.

Issues: none
Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/103528)
<!-- Reviewable:end -->

Commit:
server: Revert `server.shutdown.jobs_wait` to 0

Revert default setting for `server.shutdown.jobs_wait` to 0
to ensure that shutdown dows not wait for active jobs.

Release note: None

miretskiy commented:
bors r+
